### ISSUE TITLE
Display mass not radius

### ISSUE THREAD

**Issue by henopied** [SNSA/NEXT#30]
At the moment your extension is displaying blob radius instead of the actual mass.

**Comment by heroandtn3** [comment]
Thanks, we will fix soon.

**Comment by henopied** [comment]
btw is there a gitter for this repo?

**Comment by heroandtn3** [comment]
Currently no, but I think it's necessary to create one.

**Comment by henopied** [comment]
@heroandtn3 I fixed this and a _bunch_ of other issues should i make a pr or do you want me to create a gist first?

**Comment by heroandtn3** [comment]
@henopied great news! I think it's nice to have one PR for each issues, please do not try to resolve multiple issues in one PR or one commit.

**Comment by henopied** [comment]
@heroandtn3 your latest pr made it difficult for me to do that. I prefer having one file as it wasn't even very big.

**Comment by heroandtn3** [comment]
Sorry about that, you are right, it was not very big, but it will become big and cause many conflicts if we keep modifying on 1 file :P

**Comment by henopied** [comment]
The place that this was taken from specifically made a commit to keep it in one file because of the clutter this causes

**Comment by heroandtn3** [comment]
Don't worry much because I just splitted file but not change anything to the code, it should be easy to resolve conflicts.

**Comment by henopied** [comment]
I am also not a big fan of having to deal with commit titles and such. I guess I should get working. :P

**Comment by henopied** [comment]
The thing is though I made so many changes its hard to keep track of

**Comment by henopied** [comment]
I can't work on this for a while so if you wish to make the commits yourself this is the gist for the [main.js](https://gist.github.com/henopied/43df7b5aee341c8aba0b) and [servers.js](https://gist.github.com/henopied/d4bb32ed718441473765) also I deleted the connector file.

**Comment by heroandtn3** [comment]
OK, I will take a look.

**Comment by henopied** [comment]
Also if you wouldn't mind putting it back into 1 file for the time being for easy development. I see no need for it to be split into 7 files of ~50 lines I could see splitting it onto ~3 files when there is a lot more going on but at the moment I don't think it is necessary and it will only make it harder for development.

**Comment by heroandtn3** [comment]
@henopied I think no, we are switching to ES6 as you can see in #20 and #37, it's nice to have each class per file so it will be easy to develop and test in the furture. Sorry but I do not see any point that using 1 file makes it easier to develop. If you have trouble of conflict, please send small PRs to help us avoid conflict and review/merge your PRs easier.

BTW, I'm on my holiday so won't code much, I will see all of this when I come back several days later.

**Comment by heroandtn3** [comment]
@henopied I just made a patch file for your work, this may help you to split your work easier https://gist.github.com/heroandtn3/04c2ded290ff8e032b31

**Comment by henopied** [comment]
Thank you very much for that, but that patch still has a few memory leaks mainly
```
this.ball.on('destroy', function (reason) {
...
        setTimeout(function () {
          ...
          delete _this.main.balls[_this.ball.id];
        }, 150);
```
that one which fixes the memory leak you described. The other one I made myself, move the `spectateFieldUpdate` event listener to the constructor for the viewer.

**Comment by heroandtn3** [comment]
@henopied please consider send one PR for each patch you want to apply. It's nice if you could send a PR for memory leak soon. Just create a branch name "NEXT-30" (30 is the issue number) then commit on it.

**Comment by henopied** [comment]
Ok, I was wondering what the naming convention was for prs.

**Comment by heroandtn3** [comment]
Fixed by #85
